pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotInstanceRestletResource.java

```diff
@@ -108,6 +108,19 @@ public Instances getAllInstances() {
     return new Instances(_pinotHelixResourceManager.getAllInstances());
   }
 
+  @GET
+  @Path("/liveinstances")
+  @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_INSTANCE)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "List all live instances")
```

Comment:
Would this also list instances that have temporarily disconnected from Helix? (due to GC, regular restart, etc.)

If not, might be good to call it out explicitly in the description.

Reply:
I tested via the following:
- Restarted a node in one of our clusters. During the restart we did not get the instance id of that node in the `/liveinstances` ZK path. Post restart, we got the instance back.